consensus/consortium/v2/consortium.go

```diff
@@ -570,15 +640,37 @@ func (c *Consortium) Prepare(chain consensus.ChainHeaderReader, header *types.He
 		if err != nil {
 			return err
 		}
-		// Sort validators by address
-		sort.Sort(validatorsAscending(newValidators))
-		for _, validator := range newValidators {
-			header.Extra = append(header.Extra, validator.Bytes()...)
+
+		var (
+			blsPublicKeys       []*blst.PublicKey = make([]*blst.PublicKey, len(newValidators))
+			checkpointValidator []finality.ValidatorWithBlsPub
+		)
+
+		// Before Shillin, when encoding the checkpointExtra into header's extra data,
+		// we ignore the blsPublicKeys. So just leave the checkpointExtra's blsPublicKeys
+		// filled with zero value before Shillin.
+		if isShillin {
+			// TODO: Get finality voter public key from contract
+			// blsPublicKeys =
```

Comment:
TODO: We need to filter out the validator without BLS public key or with a malformed one